portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -31,11 +35,23 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-    ArrayList<DataComment> dataComments = new ArrayList<DataComment>();
 
     @Override
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-        // Send the JSON as the response
+        Query query = new Query("DataComment").addSort("dateCreated", SortDirection.DESCENDING);
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        PreparedQuery results = datastore.prepare(query);
+        List<DataComment> dataComments = new ArrayList<>();
+        for (Entity entity : results.asIterable()) {
+            long id = entity.getKey().getId();
+            String message = (String) entity.getProperty("message");
+            String creator = (String) entity.getProperty("creator");
+            String dateCreated = (String) entity.getProperty("dateCreated");
+            
+            DataComment dataComment = new DataComment(id, creator, message, dateCreated);
+            dataComments.add(dataComment);
+        }
```

Comment:
consider using a java stream
https://www.geeksforgeeks.org/stream-in-java/
https://howtodoinjava.com/java8/stream-map-example/

something like

```
List<DataComment> dataComments = results.asIterable()
    .map(entity -> {
        long id = entity.getKey().getId();
        String message = (String) entity.getProperty("message");
        String creator = (String) entity.getProperty("creator");
        String dateCreated = (String) entity.getProperty("dateCreated");
        
        return new DataComment(id, creator, message, dateCreated);
    })
    .collect(Collectors.toList());

```

Reply:
Working on this - I need to take a look at the syntax because I keep getting compilation errors.

Reply:
Done!